src/components/views/rooms/RoomList.js

```diff
@@ -614,23 +631,37 @@ module.exports = React.createClass({
         this._layoutSections = [];
         const defaultProps = {
             collapsed: this.props.collapsed,
-            isFiltered: !!this.props.searchFilter,
+            isFiltering: !!this.props.searchFilter,
             incomingCall: this.state.incomingCall,
         };
 
-        subListsProps.forEach((p) => {
-            p.list = this._applySearchFilter(p.list, this.props.searchFilter);
-        });
+        function visibleCount(props, count) {
+            return count + (props.extraTiles ? props.extraTiles.length : 0);
+        }
 
-        subListsProps = subListsProps.filter((props => {
-            const len = props.list.length + (props.extraTiles ? props.extraTiles.length : 0);
-            return len !== 0 || props.onAddRoom;
-        }));
+        function isSubListVisible(props, count = props.list.length) {
+            return visibleCount(props, count) !== 0 || props.onAddRoom;
+        }
+
+        subListsProps = subListsProps.filter((props) => isSubListVisible(props));
+
+        subListsProps.forEach((props) => {
+            const {count, filteredFlags} = this._applySearchFilter(props.list, this.props.searchFilter);
```

Comment:
`_mapSubListProps` is called on every `render`, which then calls the relatively CPU intensive `_applySearchFilter`, making this the likely source of Matthew's problems. Caching the result of `_applySearchFilter` (or better yet: pre-calculating it ahead of a render) would yield better results from the `render` at least, and probably make the experience nicer for users.

Reply:
alternatively, cache the whole `_mapSubListProps` to avoid all the hits